builder/bundles.go

```diff
@@ -420,24 +425,30 @@ func resolvePackages(numWorkers int, set bundleSet, packagerCmd []string, emptyD
 	return &bundleRepoPkgs, err
 }
 
-func installFilesystem(chrootDir string, localPath string, packagerCmd []string, downloadRetries int) error {
+func installFilesystem(chrootDir string, packagerCmd []string, downloadRetries int, dnfDownloadDir string, repoInfo map[string]repoInfo) error {
 	var rpmFull string
 	var err error
 
-	rpmFull = filepath.Join(localPath, fileSystemRpm)
-	rpmMap[fileSystemRpm] = true
+	rpmName := fileSystemMetadata.name + "-" + fileSystemMetadata.version + "." + fileSystemMetadata.arch + ".rpm"
+	rpmMap[rpmName] = true
+
+	if !repoInfo[fileSystemMetadata.repo].repoOnline {
+		rpmFull = filepath.Join(repoInfo[fileSystemMetadata.repo].repoURL, rpmName)
+	} else {
+		packagerCmdNew := merge(packagerCmd, "--destdir", dnfDownloadDir)
+		_, err := downloadRpms(packagerCmdNew, []string{fileSystemMetadata.name}, chrootDir, downloadRetries)
+		if err != nil {
+			return err
+		}
+		rpmFull = filepath.Join(dnfDownloadDir, rpmName)
+	}
```

Comment:
So as I mentioned in the above comment, we could determine the rpm download dir outside the queryRpmName().
E.g.
```
	rpm := pkgInfo.name + "-" + pkgInfo.version + "." + pkgInfo.arch + ".rpm"

	if repos[pkgInfo.repo].urlScheme == "file" {
		rpmDir = repos[pkgInfo.repo].url
	} else {
		rpmDir = dnfDownloadDir
		packagerCmdNew := merge(packagerCmd, "--destdir", rpmDir)
		_, err := downloadRpms(packagerCmdNew, []string{pkgInfo.name}, chrootDir, downloadRetries)
		if err != nil {
			return err
		}
	}

	rpmFull = filepath.Join(rpmDir, rpm)

		// If rpm is not found, it could be because rpm filename is not in autospec generated format.
		// In this case, query the dnf database to get the name of the rpm file containing the package.
		rpm = queryRpmName(packagerCmd, pkgInfo.name, pkgInfo.repo)
		if rpm == "" {
			return fmt.Errorf("rpm not found for pkg: %s", pkgInfo.name)
		}
		rpmFull = filepath.Join(rpmDir, rpm)
		if _, err = os.Stat(rpmFull); os.IsNotExist(err) {
			return fmt.Errorf("rpm not found for pkg: %s", pkgInfo.name)
		}
	}

	rpmMap[rpm] = true
```